projects/gloo/pkg/translator/route_config.go

```diff
@@ -429,6 +429,17 @@ func (h *httpRouteConfigurationTranslator) setWeightedClusters(params plugins.Ro
 			return err
 		}
 
+		//Catch when Customers pass weights less than 1 or forget to pass weights - this leads to envoy errors
+		if weightedDest.GetWeight() < 1 {
+			weightedDestUpstream := weightedDest.GetDestination().GetUpstream()
+
+			validation.AppendRouteError(routeReport,
+				validationapi.RouteReport_Error_ProcessingError,
+				fmt.Sprintf("Incorrect configuration for Weighted Destination for upstream: %s - Weighted Destinations require a weight that is greater than 1", weightedDestUpstream.Name),
```

Comment:
```suggestion
				fmt.Sprintf("Incorrect configuration for Weighted Destination for upstream: %s - Weighted Destinations require a weight that is greater than 0", weightedDestUpstream.Name),
```
should this say "greater than 0"? or "at least 1"?

Reply:
For integer values those two statements mean the same. I worded it this way as it captures the case where someone has passed 0 as a weight. It better calls out the situation they put themselves in.